pyodide_http/_core.py

```diff
@@ -29,13 +32,38 @@ class Response:
     body: bytes
 
 
-def send(request: Request) -> Response:
+_SHOWN_WARNING=False
+
+def send(request: Request, stream: bool = False) -> Response:
+    if request.params:
+        from js import URLSearchParams
+        params = URLSearchParams.new()
+        for k, v in request.params.items():
+            params.append(k, v)
+        request.url += "?"+params.toString()
+
+    global _SHOWN_WARNING
     from js import XMLHttpRequest
     try:
         from js import importScripts
         _IN_WORKER = True
     except ImportError:
         _IN_WORKER = False
+    # support for streaming workers (in worker )
+    if stream:
+        if not _IN_WORKER:
+            stream = False
+            if not _shown_warning:
```

Comment:
Please move the warning code to a separate function so it can be used from `_streaming.py` too. Those files contain very similar code right now.

Reply:
good idea, done